apps/email/js/sync.js

```diff
@@ -336,28 +341,40 @@ define(function(require) {
             };
 
             if (model.getAccountCount() === 1) {
-              subject = info.subject;
-              body = info.from;
+              subjectL10n = { raw: info.subject };
+              bodyL10n = { raw: info.from };
             } else {
-              subject = mozL10n.get('new-emails-notify-multiple-accounts', {
-                n: count,
-                accountName: result.address
-              });
-              body = mozL10n.get('new-emails-notify-multiple-accounts-body', {
-                from: info.from,
-                subject: info.subject
-              });
+              subjectL10n = {
+                id: 'new-emails-notify-multiple-accounts',
+                args: {
+                  n: count,
+                  accountName: result.address
+                }
+              };
+              bodyL10n = {
+                id: 'new-emails-notify-multiple-accounts-body', 
+                args: {
+                  from: info.from,
+                  subject: info.subject
+                }
+              };
             }
           }
 
-          sendNotification(
-            result.id,
-            subject,
-            body,
-            iconUrl,
-            dataObject,
-            behavior
-          );
+          if (!bodyL10n.then) {
+            bodyL10n = Promise.resolve(bodyL10n);
+          }
+          
+          bodyL10n.then(() => {
```

Comment:
You need to take the result of the promise:
```javascript
bodyL10n.then(body => {
  sendNotification(
    ...,
    body,
    ...,
  );
});
```

otherwise you are passing a Promise.